### ISSUE TITLE
explorer.compactFolders is still broken for screen reader users

### ISSUE THREAD

**Issue by Neurrone** [microsoft/vscode#107235]
Issue Type: <b>Bug</b>

/CC @isidorn

Previously reported in #93465, but it is now locked.

1. Attempt to navigate explorer view with NVDA with compact folders enabled (which it is by default).

Expected: all items read correctly.
Actual: when you have a/b/c.txt, in some instances NVDA skips out reading a when arrowing to it and reads folder b instead, with no real indication that its happening.

This is really confusing for screen reader users. Until it is fixed, I think its much easier to just default it to false as a temporary workaround.

VS Code version: Code - Insiders 1.49.0-insider (00754e4babebc9ee5aa881234bc16366c4f32aa9, 2020-09-02T05:56:03.393Z)
OS version: Windows_NT x64 10.0.19041

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz (8 x 1992)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.86GB (7.72GB free)|
|Process Argv|--crash-reporter-id a6913056-d129-47e1-a262-b9f028a87202|
|Screen Reader|yes|
|VM|20%|
</details><details><summary>Extensions (18)</summary>

Extension|Author (truncated)|Version
---|---|---
notes|can|1.0.1
vscode-markdownlint|Dav|0.36.3
vscode-eslint|dba|2.1.8
gitlens|eam|10.2.2
prettier-vscode|esb|5.6.0
foam-vscode|foa|0.3.1
go|gol|0.17.0
rest-client|hum|0.24.2
vscode-markdown-notes|kor|0.0.19
python|ms-|2020.8.109390
remote-wsl|ms-|0.44.5
cpptools|ms-|1.0.1
vsnotes|pat|0.7.1
rust|rus|0.7.8
markdown-preview-enhanced|shd|0.5.13
markdown-links|tch|0.8.0
vscode-webhint|web|1.5.8
markdown-all-in-one|yzh|3.3.0

</details>

**Comment by isidorn** [comment]
Thanks for letting us know, however I just tried this out with latest VS Code insiders and this works just fine for me.
If I understand your description correctly: you do not like the fact that the explorer remembers what was the last active subnode.

So have `a / b / c`, and focus onto `b` via the keyboard, now press `down` some other element is focused. Now press `up` `b` is focused again - since it was the last focused subnode. But you do not like that, you would like `a` to be focused instead?
Am I correct in inferring the problem you are facing? Or is it something else?

**Comment by Neurrone** [comment]
Say i have the following folder structure

```
.vscode
assets
	css
		style.css	
```

Lets say that the state of the explorer view has the vscode folder collapsed and assets expanded so that it shows the css folder. Then, pressing down arrow when focus is on .vscode moves it to assets and the following is heard: "level 2 css expanded". The critical context missing here is the fact that this is actually under the assets folder, which is not announced and for me at least, causes a lot more confusion. I then have to figure out where I actually am, e.g by pressing left arrow to go up one level so it says "assets expanded".

Does that make sense?

In short, this feature might not really be intuitive from the screen reader user's perspective because of the lost information during navigation.

**Comment by Neurrone** [comment]
I've also found a bug where css and style.css when they appear in the tree, are both announced as level 2, which isn't semantically accurate.

**Comment by isidorn** [comment]
@Neurrone I have just pushed a fix for your latest comment - the aria-level computation should now be fine for all cases.
I am still thinking about your first comment, will tackle it soon probably

**Comment by isidorn** [comment]
@neurrone How about the following solution: we change the aria label of the subnodes to be the following:
`NAME compact parent1 / parent 2 / parent 3... /name`

So in your example the aria label would be
"level 2 css compact assets/css expanded"

If you have a better solution for the aria lable of a sub node I am open for suggestions
